actions/pr-quality-check/action.yml

```diff
@@ -0,0 +1,1033 @@
+            echo "Context files by rule:"
+            # Print rule -> files (no reason)
+            printf '%s\n' "$PER_RULE_CONTEXT_JSON_COMPACT" | jq -r 'to_entries[] | "  [\(.key)] -> \(.value.context_files | join(", "))"'
+          else
+            echo "Context files by rule: none"
+          fi
+          if [ ${ERRORS_IN_FILE:-0} -eq 0 ] && [ ${WARNINGS_IN_FILE:-0} -eq 0 ]; then
+            echo "Outcome: PASS"
+          else
+            echo "Outcome: ${ERRORS_IN_FILE:-0} errors, ${WARNINGS_IN_FILE:-0} warnings"
+
+          fi
+          echo " ------------------------------------------------------------"
+          echo " ------------------------------------------------------------"
+        done <<< "$CHANGED_FILES"
+
+        # Add summary
+        echo "## 📊 Summary" >> quality_results.md
+        echo "" >> quality_results.md
+        echo "- **Files changed in PR**: $TOTAL_CHANGED_FILES" >> quality_results.md
+        echo "- **Files analyzed**: $TOTAL_FILES_CHECKED (matched quality check patterns)" >> quality_results.md
+        echo "- **Issues found**: $TOTAL_ERRORS" >> quality_results.md
+        echo "- **Warnings**: $TOTAL_WARNINGS" >> quality_results.md
+        echo "" >> quality_results.md
+
+        TOTAL_ISSUES=$((TOTAL_ERRORS + TOTAL_WARNINGS))
+        if [ $TOTAL_ISSUES -eq 0 ]; then
+          echo "✅ **All checks passed!** Your code meets the quality standards." >> quality_results.md
+        else
+          if [ $TOTAL_ERRORS -gt 0 ]; then
+            echo "❌ **Please address the issues above before merging.**" >> quality_results.md
+          else
+            echo "⚠️ **Some warnings found.** Consider addressing them to improve code quality." >> quality_results.md
+          fi
+        fi
+
+        echo "" >> quality_results.md
+        echo "---" >> quality_results.md
+        echo "*Generated by PR Quality Check* 🤖 - Comment updated with every commit (last refresh: ${HEAD_SHA_SHORT}) " >> quality_results.md
+
+        echo "TOTAL_CHANGED_FILES=$TOTAL_CHANGED_FILES" >> $GITHUB_ENV
+        echo "TOTAL_ERRORS=$TOTAL_ERRORS" >> $GITHUB_ENV
+        echo "TOTAL_WARNINGS=$TOTAL_WARNINGS" >> $GITHUB_ENV
+        echo "TOTAL_FILES_CHECKED=$TOTAL_FILES_CHECKED" >> $GITHUB_ENV
+
+    - name: Post quality check results
+      if:
+        env.SHOULD_RUN == 'true' && env.HAS_MATCHING_CHANGES == 'true' &&
+        inputs.post-comment == 'true'
+      shell: bash
+      env:
+        GH_TOKEN: ${{ inputs.gh-token }}
+        PR_NUMBER: ${{ github.event.pull_request.number }}
+        REPO: ${{ github.repository }}
+      run: |
+        echo "💬 Posting quality check results to PR #$PR_NUMBER"
+
+        # Check if we already have a comment from this action
+        EXISTING_COMMENT=$(gh api repos/$REPO/issues/$PR_NUMBER/comments --jq '.[] | select(.body | contains("Generated by PR Quality Check")) | .id' | head -1)
```

Comment:
If you want you could fingerprint the PR comment with an invisible HTML comment in the body. Might catch some edge cases.

For example: https://github.com/smartcontractkit/.github/blob/68f9f07e37146c8c1555c191e72032078204a384/actions/apidiff-go/src/github.ts#L4